### PR TITLE
Bring your pet to work day!

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
Adds a new neutral station trait where people are able to bring their personal pets to show off to the rest of the crew (or to donate to the chef). Before joining, players will be able to customize their pets

https://github.com/user-attachments/assets/5a5cda18-e551-4f87-b649-67a6940ffc6a

u can set their name, gender, pet carrier color, and  teach them give them a unique dance sequence trick. if players dont wish to opt in, they also have the choice not to

## Why It's Good For The Game
adds a new station trait which people can RP around, whether its by competitively comparing pets, or resolving conflicts around pets being killed/kidnapped. Ive also GAGsified pet carriers, so people can now have the option to recolor those too

## Changelog
:cl:
add: adds a new station trait, "bring your pet to work" day
/:cl:

### CODE REVIEW

File: code/modules/mob/living/basic/pets/pet_designer.dm [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	return pet_options
+
+/datum/pet_customization/ui_act(action, params, datum/tgui/ui)
+	. = ..()
+	switch(action)
+		if("finalize_pet")
+
+			var/pet_type = text2path(params["selected_path"])
+			if(ispath(pet_type, /mob/living/basic/pet))
+				selected_path = pet_type
```
Comment (MrMelbert) [inline code/modules/mob/living/basic/pets/pet_designer.dm:129]: Sanity check that the passed pet type is one of the valid pets

File: code/__DEFINES/basic_mobs.dm [hunk 2]
```diff
@@ -33,6 +33,8 @@
 ///keeps track of how many gutlunches are born
 GLOBAL_VAR_INIT(gutlunch_count, 0)
 
+GLOBAL_LIST_EMPTY(customized_pets)
```
Comment (MrMelbert) [inline code/__DEFINES/basic_mobs.dm:36]: Needs an auto doc

**Comment by Admiral-Gigaton** [comment]
"magnumopus" is the best name for a little dance trick. best PR showing of the year.

**Comment by DaydreamIQ** [comment]
Humbly requesting a pet mouse as an option for all the rodent loving homies out there

**Comment by necromanceranne** [comment]
In contrast to the mouse request, what about duct spiders :3

**Review by Wallemations — APPROVED** [review]
Cute!

**Comment by SomeoneYouProbablyKnow** [comment]
1. I'd be nice to have a random option.
2. This makes it possible to bring a Corgi named Ian and replace the actual Ian with this. I think this is a very funny interaction but I am bringing attention to it still
3. Parrots, need parrots

**Comment by Loafin34** [comment]
1. exotic corgi
2. sholean grapes
3. blob spore
please

**Comment by krookodilehunter** [comment]
puppies and kittens too?

**Comment by Y0SH1M4S73R** [comment]
what about false carp (the ones spawned by the shuttle event)

**Comment by OneAsianTortoise** [comment]
that's a dog

**Comment by theselfish** [comment]
361606158-68166997-cc4d-46a4-ac26-be2e484f4460

**Comment by theselfish** [comment]
> 1. I'd be nice to have a random option.
> 
>     2. This makes it possible to bring a Corgi named Ian and replace the actual Ian with this. I think this is a very funny interaction but I am bringing attention to it still
> 
>     3. Parrots, need parrots

If someone Ship of Theseus' Ian, they deserve an award.

**Comment by Ben10Omintrix** [comment]
https://github.com/user-attachments/assets/2479e33f-c82e-4e33-92ea-59fde1e667a3

ive added some new pets based on the requests from here

**Comment by OneAsianTortoise** [comment]
> newpets.mp4 
> ive added some new pets based on the requests from here

where turtle

**Review by Wallemations — CHANGES_REQUESTED** [review]
If you want the GAGS colors to work best for the underside, remove the blue tint & make it purely white & grey. That way the color is 100% the one you pick.

**Comment by TheSmallBlue** [comment]
Frog! Frog! Put frog under vermins please!!

**Comment by Admiral-Gigaton** [comment]
lizard pet? crabs? penguin?

**Comment by Twaticus** [comment]
how does a pet have some alternate gender lol

**Comment by iliyaxox** [comment]
> how does a pet have some alternate gender lol

youll see once i bring my non binary cat to the station

**Comment by Loafin34** [comment]
cockroach please

**Pull request merged**